src/compiler/rename.rs

```diff
@@ -38,44 +35,38 @@ fn rename_in_cons(namemap: &HashMap<Vec<u8>, Vec<u8>>, body: Rc<SExp>) -> Rc<SEx
             None => body,
         },
         SExp::Cons(l, f, r) => {
-            match f.borrow() {
-                SExp::Atom(la, q) => {
-                    if *q == "q".as_bytes().to_vec() {
-                        return Rc::new(SExp::Cons(
-                            l.clone(),
-                            Rc::new(SExp::Atom(la.clone(), "q".as_bytes().to_vec())),
-                            r.clone(),
-                        ));
-                    } else if *q == "quote".as_bytes().to_vec() {
-                        return r
-                            .proper_list()
-                            .map(|x| match &x[..] {
-                                [v] => Rc::new(SExp::Cons(
-                                    l.clone(),
-                                    Rc::new(SExp::atom_from_string(
-                                        la.clone(),
-                                        &"quote".to_string(),
-                                    )),
-                                    Rc::new(SExp::Cons(
-                                        v.loc(),
-                                        Rc::new(v.clone()),
-                                        Rc::new(SExp::Nil(v.loc())),
-                                    )),
+            if let SExp::Atom(la, q) = f.borrow() {
+                if *q == "q".as_bytes().to_vec() {
+                    return Rc::new(SExp::Cons(
+                        l.clone(),
+                        Rc::new(SExp::Atom(la.clone(), "q".as_bytes().to_vec())),
+                        r.clone(),
+                    ));
+                } else if *q == "quote".as_bytes().to_vec() {
```

Comment:
I would think you could change this string literal to just be `b"quote"`, as well as the "q" above.